### PR TITLE
Paper frames, but with new sprites.

### PR DESCRIPTION
**Pull request by Expletive**
:cl:
add: Paper frames can be created using wood and paper, and can be used to create decorative structures.
add: Natural paper can now be crafted 
add: KorPhaeron can now start a maid cafe, if they so desire.
/:cl:

I gave spriting another shot, since I was asked to. These sprites smooth properly, but are probably still crap. I opened a new PR, since the old one had things I've already atomized out into PRs that have since been merged.

scrnshot1

### CODE REVIEW

**Comment by QualityVan** [comment]
horizontal door looks weeeeeeird

**Comment by bgobandit** [comment]
I'm confused about what these are for

**Comment by KorPhaeron** [comment]
It says in the changelog

**Comment by bgobandit** [comment]
Yeah but the photo doesn't really explain what "decorative structures" are or why this would make a maid cafe

**Comment by KorPhaeron** [comment]
https://upload.wikimedia.org/wikipedia/commons/e/e1/Takamatsu-Castle-Building-Interior-M3488.jpg

**Comment by Repukan** [comment]
Slibbery slobe

**Comment by WJohn** [comment]
These are much improved, good work.

Not a fan of the horizontal doors, but that's mostly since our doors don't do that, they're all the same direction.

**Comment by dannno** [comment]
Kor is not in charge of the Maid Cafes, so if you could please fix your changelog to say kor is a stupid idiot

**Comment by Alek2ander** [comment]
flammable - :white_check_mark: 
:ok_hand:

**Comment by nicbn** [comment]
They look too thick, maybe make it 1-dir like the thin windows?

**Comment by Iatots** [comment]
They shouldn't smooth, the whole point is that they are made of panels. It would be better for the doors too, since without smoothing you could have a wall flip 90° with a wrench and screwdriver.

**Comment by octareenroon91** [comment]
Paper windows seem to be transparent. Why are they transparent?

Also I think 'divider' or 'room divider' might have more common usage than 'separator'

**Comment by Bawhoppen** [comment]
we need something that allows light to pass but is still dense you're right

**Comment by Expletive** [comment]
@octareenroon91  They are not transparent. I just used xray eyes fr the screenshot.

File: code/game/objects/items/stacks/sheets/sheet_types.dm [hunk 1]
```diff
@@ -397,3 +398,26 @@ GLOBAL_LIST_INIT(plastic_recipes, list(
 /obj/item/stack/sheet/plastic/Initialize(mapload, new_amount, merge = TRUE)
 	recipes = GLOB.plastic_recipes
 	. = ..()
+
+GLOBAL_LIST_INIT(paperframe_recipes, list(
+new /datum/stack_recipe("paper frame separator", /obj/structure/window/paperframe, 2, one_per_turf = 1, on_floor = 1, time = 10), \
+new /datum/stack_recipe("paper frame door", /obj/structure/mineral_door/paperframe, 3, one_per_turf = 1, on_floor = 1, time = 10 )))
```
Comment (Cyberboss) [inline code/game/objects/items/stacks/sheets/sheet_types.dm]: TRUE for one_per_turf and on_floor

File: code/game/objects/structures/mineral_doors.dm [hunk 2]
```diff
@@ -87,10 +87,10 @@
 /obj/structure/mineral_door/proc/Open()
 	isSwitchingStates = 1
 	playsound(loc, openSound, 100, 1)
+	set_opacity(0)
```
Comment (Cyberboss) [inline code/game/objects/structures/mineral_doors.dm]: FALSE

File: code/game/objects/structures/mineral_doors.dm [hunk 3]
```diff
@@ -110,7 +110,7 @@
 	flick("[initial_state]closing",src)
 	sleep(10)
 	density = 1
-	opacity = 1
+	set_opacity(1)
```
Comment (Cyberboss) [inline code/game/objects/structures/mineral_doors.dm]: TRUE

File: code/game/objects/structures/mineral_doors.dm [hunk 4]
```diff
@@ -179,7 +179,7 @@
 
 /obj/structure/mineral_door/transparent/Close()
 	..()
-	opacity = 0
+	set_opacity(0)
```
Comment (Cyberboss) [inline code/game/objects/structures/mineral_doors.dm]: FALSE

File: code/game/objects/structures/mineral_doors.dm [hunk 5]
```diff
@@ -219,3 +219,22 @@
...
+	obj_integrity = 20
+	max_integrity = 20
+
+/obj/structure/mineral_door/paperframe/Initialize()
+	. = ..()
+	queue_smooth_neighbors(src)
+
+/obj/structure/mineral_door/paperframe/deconstruct(disassembled = TRUE)
+	..()
+	queue_smooth_neighbors(src)
```
Comment (Cyberboss) [inline code/game/objects/structures/mineral_doors.dm]: Shouldn't this be Destroy() instead?

File: code/game/objects/structures/window.dm [hunk 6]
```diff
@@ -181,7 +183,7 @@
 				playsound(loc, WT.usesound, 40, 1)
 				if(do_after(user, 40*I.toolspeed, target = src))
 					obj_integrity = max_integrity
-					playsound(loc, 'sound/items/welder2.ogg', 50, 1)
+					playsound(loc, 'sound/items/Welder2.ogg', 50, 1)
```
Comment (Cyberboss) [inline code/game/objects/structures/window.dm]: playsound(src, ...

File: code/game/objects/structures/window.dm [hunk 7]
```diff
@@ -157,7 +159,7 @@
 	user.changeNext_move(CLICK_CD_MELEE)
 	user.visible_message("[user] knocks on [src].")
 	add_fingerprint(user)
-	playsound(loc, 'sound/effects/glassknock.ogg', 50, 1)
+	playsound(loc, 'sound/effects/Glassknock.ogg', 50, 1)
```
Comment (Cyberboss) [inline code/game/objects/structures/window.dm]: playsound(src, ...

File: code/game/objects/structures/window.dm [hunk 8]
```diff
@@ -144,7 +146,7 @@
 	user.changeNext_move(CLICK_CD_MELEE)
 	user.visible_message("<span class='notice'>Something knocks on [src].</span>")
 	add_fingerprint(user)
-	playsound(loc, 'sound/effects/glassknock.ogg', 50, 1)
+	playsound(loc, 'sound/effects/Glassknock.ogg', 50, 1)
```
Comment (Cyberboss) [inline code/game/objects/structures/window.dm]: playsound(src, ...

File: code/game/objects/structures/window.dm [hunk 9]
```diff
@@ -599,3 +600,93 @@
...
+	armor = list(melee = 0, bullet = 0, laser = 0, energy = 0, bomb = 0, bio = 0, rad = 0, fire = 0, acid = 0)
+	breaksound = 'sound/items/poster_ripped.ogg'
+	hitsound = 'sound/weapons/slashmiss.ogg'
+	var/static/mutable_appearance/torn
+	var/static/mutable_appearance/paper
+
+/obj/structure/window/paperframe/Initialize()
+	. = ..()
+	torn = mutable_appearance('icons/obj/smooth_structures/paperframes.dmi',icon_state = "torn", layer = ABOVE_OBJ_LAYER - 0.1)
+	paper = mutable_appearance('icons/obj/smooth_structures/paperframes.dmi',icon_state = "paper", layer = ABOVE_OBJ_LAYER - 0.1)
```
Comment (Cyberboss) [inline code/game/objects/structures/window.dm]: Assign these at their definition

File: code/game/objects/structures/window.dm [hunk 10]
```diff
@@ -599,3 +600,93 @@
...
+	for(var/obj/item/I in debris)
+		debris -= I
+		qdel(I)
```
Comment (Cyberboss) [inline code/game/objects/structures/window.dm]: if debris only contains items, just do QDEL_LIST(debris)

File: code/game/objects/structures/window.dm [hunk 11]
```diff
@@ -599,3 +600,93 @@
...
+
+	var/papers = rand(1,4)
+	debris += new /obj/item/stack/sheet/mineral/wood()
+	for(var/i in 1 to papers)
+		debris += new /obj/item/weapon/paper/natural()
+	update_icon()
+
+/obj/structure/window/paperframe/attack_hand(mob/user)
+	if(!can_be_reached(user))
+		return
```
Comment (Cyberboss) [inline code/game/objects/structures/window.dm]: Remove this check, attack_hand already verifies that before it's called

File: code/game/objects/structures/window.dm [hunk 12]
```diff
@@ -599,3 +600,93 @@
...
+	add_fingerprint(user)
+	if(user.a_intent != INTENT_HARM)
+		user.changeNext_move(CLICK_CD_MELEE)
+		user.visible_message("[user] knocks on [src].")
+		playsound(loc, "pageturn", 50, 1)
```
Comment (Cyberboss) [inline code/game/objects/structures/window.dm]: playsound(src, ...

File: code/game/objects/structures/window.dm [hunk 13]
```diff
@@ -599,3 +600,93 @@
...
+	else
+		take_damage(4,BRUTE,"melee", 0)
+		playsound(loc, hitsound, 50, 1)
```
Comment (Cyberboss) [inline code/game/objects/structures/window.dm]: playsound(src, ...

File: code/game/objects/structures/window.dm [hunk 14]
```diff
@@ -599,3 +600,93 @@
...
+		if(!QDELETED(src))
+			user.visible_message("<span class='danger'>[user] tears a hole in [src].</span>")
+			update_icon()
+
+/obj/structure/window/paperframe/update_icon()
+	if(obj_integrity < max_integrity)
+		cut_overlay(paper)
+		add_overlay(torn)
+		set_opacity(0)
```
Comment (Cyberboss) [inline code/game/objects/structures/window.dm]: FALSE

File: code/game/objects/structures/window.dm [hunk 15]
```diff
@@ -599,3 +600,93 @@
...
+	else
+		cut_overlay(torn)
+		add_overlay(paper)
+		set_opacity(1)
```
Comment (Cyberboss) [inline code/game/objects/structures/window.dm]: TRUE

File: code/game/objects/structures/window.dm [hunk 16]
```diff
@@ -599,3 +600,93 @@
...
+	queue_smooth(src)
+
+
+/obj/structure/window/paperframe/attackby(obj/item/weapon/W, mob/user)
+	if(istype(W, /obj/item/weapon/weldingtool))
+		var/obj/item/weapon/weldingtool/WT = W
+		if(WT.isOn())
+			fire_act(W.is_hot())
```
Comment (Cyberboss) [inline code/game/objects/structures/window.dm]: I think weldingtool/is_hot returns FALSE if it's off. Remove the weldingtool specific stuff

File: code/game/objects/structures/window.dm [hunk 17]
```diff
@@ -599,3 +600,93 @@
...
+		return
+	else if(W.is_hot())
+		fire_act(W.is_hot())
+		return
+	if(user.a_intent == INTENT_HARM)
+		return ..()
+	else
+		if(istype(W, /obj/item/weapon/paper) && obj_integrity < max_integrity)
```
Comment (Cyberboss) [inline code/game/objects/structures/window.dm]: Make these 2 lines 1

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by LarissaMayrink** [comment]
>maid cafe barsigh added
>maid suit added to the barman wardrobe
>weeb paperwalls

**Comment by LarissaMayrink** [comment]
Its a lube covered slope, we are way beyond a slippery one

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**